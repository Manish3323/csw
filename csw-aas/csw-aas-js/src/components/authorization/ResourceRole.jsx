import React from 'react'
import PropTypes from 'prop-types'
import {TMTAuthContext} from '../TMTAuthContext.jsx'

class ResourceRole extends React.Component {
  render() {
    return (
      <TMTAuthContext.Consumer>
        {context => {
          return (
            <div className='card-content'>
              {(context.isAuthenticated && context.tmtAuth.hasResourceRole(this.props.role, this.props.resource)) ? this.props.children : null}
            </div>
          )
        }}
      </TMTAuthContext.Consumer>
    )
  }
}

ResourceRole.propTypes = {
  role: PropTypes.string.isRequired,
  resource: PropTypes.string,
  children: PropTypes.node
};

ResourceRole.contextType = TMTAuthContext;

export default ResourceRole
