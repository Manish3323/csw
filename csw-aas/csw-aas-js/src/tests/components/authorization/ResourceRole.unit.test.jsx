import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ResourceRole from '../../../components/authorization/ResourceRole'

describe('<RealmRole />', () => {
  Enzyme.configure({ adapter: new Adapter() })

  beforeEach(() => {
    jest.resetModules()
  })

  it('should render children elements if authenticated user has specified resource role', () => {
    const props = {
      resourceRole: 'person-role',
      resource: 'example-server',
      children: <div className='resource-role'>Authorization successful</div>,
      context: {
        tmtAuth: {
          hasResourceRole: jest.fn().mockImplementation(() => {
            return true
          })
        },
        isAuthenticated: true
      }
    }

    const wrapper = shallow(<ResourceRole {...props} />)

    expect(wrapper.find('div.resource-role').length).toBe(1)
  })

  it('should not render children elements if un-authenticated', () => {
    const props = {
      resourceRole: 'some-role',
      resource: 'some-server',
      children: <div className='resource-role'>Authorization successful</div>,
      context: { tmtAuth: null, isAuthenticated: false }
    }

    const wrapper = shallow(<ResourceRole {...props} />)

    expect(wrapper.find('div.resource-role').length).toBe(0)
  })

  it('should not render children elements if authenticated user does not have specified resource role', () => {
    const props = {
      resourceRole: 'person-role',
      resource: 'example-server',
      children: <div className='resource-role'>Authorization successful</div>,
      context: {
        tmtAuth: {
          hasResourceRole: jest.fn().mockImplementation(() => {
            return false
          })
        },
        isAuthenticated: true
      }
    }

    const wrapper = shallow(<ResourceRole {...props} />)

    expect(wrapper.find('div.resource-role').length).toBe(0)
  })
})
