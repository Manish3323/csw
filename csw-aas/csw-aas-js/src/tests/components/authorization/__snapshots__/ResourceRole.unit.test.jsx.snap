// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResourceRole /> should not render ResourceRole if authentication is false  1`] = `
<div
  className="error"
>
  Authentication unsuccessful
</div>
`;

exports[`<ResourceRole /> should not render ResourceRole if authentication is true but invalid Resource role 1`] = `
<div
  className="error"
>
  Authentication unsuccessful
</div>
`;

exports[`<ResourceRole /> should render ResourceRole if authentication is true and with valid Resource role 1`] = `
<div
  className="resource-role"
>
  Authentication successful
</div>
`;

exports[`<ResourceRole /> should render nothing if authentication is false and error component is not provided 1`] = `null`;
